lib/NameProvider.h

```diff
@@ -9,8 +9,11 @@ class NameProvider
 public:
     virtual ~NameProvider(void) = default;
```

Comment:
Move these includes to the default .cpp file, and instead forward declare all the various ops inside of the `vast::hl` namespace in this header.